File: Maintix_API/Models/Equipo.cs
namespace Maintix_API.Models
{
    public class Equipo
    {
        public int Id { get; set; }
        public int TipoMaquinariaId { get; set; }
        public DateTime? FechaFabricacion { get; set; }
        public string? NumeroSerie { get; set; }
        public int HorasActuales { get; set; } = 0;
        public int ContadorTipoA { get; set; } = 0;
        public int ContadorTipoB { get; set; } = 0;
        public int ContadorTipoC { get; set; } = 0;
    }
}

File: mobile/app/src/main/java/com/tonhete/maintixapp/data/models/Equipo.kt
package com.tonhete.maintixapp.data.models

import com.google.gson.annotations.SerializedName

data class Equipo(
    @SerializedName("id")
    val id: Int,

    @SerializedName("tipoMaquinariaId")
    val tipoMaquinariaId: Int,

    @SerializedName("fechaFabricacion")
    val fechaFabricacion: String?,

    @SerializedName("numeroSerie")
    val numeroSerie: String?,

    @SerializedName("horasActuales")
    val horasActuales: Int?,

    @SerializedName("tipoMaquinaria")
    val tipoMaquinaria: TipoMaquinaria?

)

data class TipoMaquinaria(
    @SerializedName("id")
    val id: Int,

    @SerializedName("descripcion")
    val descripcion: String,

    @SerializedName("proveedorId")
    val proveedorId: Int?
)

data class ActualizarHorasDto(
    @SerializedName("horasNuevas") val horasNuevas: Int
)

data class Proveedor(
    @SerializedName("id") val id: Int,
    @SerializedName("nombre") val nombre: String,
    @SerializedName("tlf") val tlf: String?,
    @SerializedName("direccion") val direccion: String?
)